src/context/announcements/announcements-context.js

```diff
@@ -0,0 +1,48 @@
+import propTypes from "prop-types";
+
+import React, { useMemo, useState } from "react";
+import { ANNOUNCEMENTS_LIST } from "../../constants/announcements";
+
+const AnnouncementsContext = React.createContext({
+  announcements: [],
+  filter: () => {},
+});
+
+export const AnnouncementsContexProvider = ({ children }) => {
+  const [filteredAnnouncements, setFilteredAnnouncements] = useState([]);
+
+  const filter = (data) => {
+    const { Phrase, Kategoria, Animals } = data;
+    let announcements = [...ANNOUNCEMENTS_LIST];
+    announcements = announcements.filter((announcement) =>
+      announcement.title.toLowerCase().includes(Phrase),
+    );
+    announcements = announcements.filter((announcement) =>
+      announcement.category.toLowerCase().includes(Kategoria),
+    );
+    announcements = announcements.filter((announcement) =>
+      announcement.animal.toLowerCase().includes(Animals),
+    );
+    setFilteredAnnouncements(announcements);
```

Comment:
You filter firstly those that have `title` equal to `Phrase`, `category` equal `Kategoria` (why not `Category`) and in the end `animal` equal `Animals`. Chain those or make a bool operation, for example `const shouldBeReturned = announcement.title.toLowerCase().includes(Phrase) && announcement.category.toLowerCase().includes(Category) && announcement.animal.toLowerCase().includes(Animals); return shouldBeReturned;`.